rightBox css 구성

sin-hyunjin opened the pull request:
## Issues 번호 : 
Closes #29 

## 변경, 추가된 코드(설명 등)
 - rightBox layout 코드 작성
 - 
 -
 -
 -
 
## 코드 주의점
 -
 -
 -
 -

sin-hyunjin committed:
rightBox css 구성

sin-hyunjin commented:
textbox 파일삭제 -> utils/string.js 파일에 코드취합하였기때문에 삭제하였음

sin-hyunjin committed:
Merge branch 'dev' into feature/#29_RightBox_Layout

sin-hyunjin committed:
Merge branch 'dev' into feature/#29_RightBox_Layout

sin-hyunjin committed:
SummaryButton right box 뼈대잡음

sin-hyunjin committed:
tailwild.css 테스트

sin-hyunjin committed:
rightBox 구현중인데 하트 버튼을 눌렀을때와 codeEditer부분이 hover되었을때 나오는 부분을 어떻게 짜야할지 reacthook부분을 더 공부해봐야할듯

sin-hyunjin committed:
summaryRightBox.js 하트버튼 클릭시 애니메이션 변경해야함

1. rightbox.css삭제 -> codeEditor.css에 추가
2. Summary.css -> summary.css 이름변경
3. rightBox.js -> summaryRightBox.js d이름변경
4. 애니메이션 집에서 공부하기

sin-hyunjin committed:
summaryRightBox 애니메이션화

1. 이제 hover시 박스 테두리가 생기는 작업을 해야할듯

doyou1 commented:
- Heart 아이콘 관련 피드백
사이즈가 안맞고, 가운데정렬이 안되어있습니다.

<img width="1710" alt="image" src="https://github.com/doyou1/web-study-clone/assets/56281493/14c432f7-15e6-4cce-8fe1-4c9902e767ac">

HeartBackGround의 svg에 정의된 클래스는 <HeartBackGround className="..."으로 빼고(위로 올리는 개념이지? svg className이 적용이 안되서 하트가 작은겨,,,), circle에 정의된 className은 tailwind을 적용하는건데, 굳이 따로 빼지말고, css파일에서 적용하는걸로 ... 부모인 button 쪽에서 isLike에 따라서 line(empty), fill(full) 클래스을 주고, .line circle { ~~~ } .flll circle { ~~~ } 요런식으로 처리하면 어떨까? 내용 이해 안되면 카톡ㄱㄱ

- hover시 박스 테두리
혼자서 생각해 봤는데, main.js 쪽에서 각각의 textbox의 hover 여부를 담는 object 상태를 만들고, 각각의 codeEditor의 prop으로 해당하는 textbox의 hover 여부 object를 전달해주고, 그 prop을 받아서 hover일 경우, hovered 클래스가 추가되서 박스 테두리를 그리는 css가 구현되도록 하면 어떨까싶음. 말이 정리가 안되는데, 
1. hover관련 이벤트가 mouseover, mouseleave 일거니까 한번 확인해보고,
2. textBox가 있는 모든 summary, summarybutton의 각각의 태그 hover 여부를 담는 reactive status를 main.js에서 관리하는 방법
3. 2의 status를 각각의 codeeditor에 집어넣고 textBox의 hover에 따라 codeeditor에 연동되도록
하는 방법이 제 머리속에 떠오릅니다. 

대충 생각나는대로 끄적여서 좀 이해하기 어려울 수도 있는데, 이해안되거나 모르겠는거 있으면 말하도록

sin-hyunjin committed:
heart.svg 변경 button 클래스.line circle 추가후 css로 애니메이션 줌

sin-hyunjin commented:
작업할때는 heart.svg가 가운데 위치하고 크기도 정상적으로 되어있었는데
why? 껐다가 다시확인하니까 크기가 작아지고 위치도 변경되어있음 다시수정필요

sin-hyunjin committed:
<HeartBackGround/> 안에 tailwind.css를 적용해야 됨

sin-hyunjin committed:
이벤트관련 로직

더 공부해야할듯
react hook 이용하면 가능할꺼같은데 만약 porps인자를 받으려면 codeEditor 코드를 객체 형식이 아니라 함수형태로 만들어야해야하나... 복잡하네

sin-hyunjin committed:
codeEditor함수형 형태로 바꾸니까 기능은 작동함 하지만 문제점이 많음

1. main.js에서 porps로 codeeditor , rightBox로 인자전달
-> 일단 mouse over 됐을때 두 구역 기능이 작동됨

문제점 1. mouseover로 하다보니 모든 구역이 선택이 됨
-> 그러면 조건문으로 해야하는데 로직구현 샘플링좀 ㅠ
1.1 클래스네임 바꿔도 안됨
1.2 파워코딩을 해보려고 여러개의 useState를 만들어도 똑같은 true, false값이다 보니 안됨
1.3 map함수랑 key값을 이용해서 사용해보니 모든 글자가 반복됨
-> 근데 이방법이 가장 가능성이 있을꺼같음
그이유는 각구역을 key,값을 만들어서 어떤구역인지 알수 있는 않을까? 라는생각

문제점 2. 반응형으로 css짜야하는데 머리가 안돌아감

sin-hyunjin committed:
codeEditer 마우스이벤트

sin-hyunjin committed:
useEffect로 결과값 출력이 가능함

sin-hyunjin committed:
마우스이벤트시 rightBox잘 작동되는거 확인

바꿀점
1. 코드를 단순화 할수 없을까?
-> 생각해봐야할듯
2. 스타일반응형으로 변경해야함

sin-hyunjin committed:
summaryRightBox.js
- useEffect 부분setstyle tailwind로 변경

sin-hyunjin committed:
Merge branch 'dev' into feature/#29_RightBox_Layout

sin-hyunjin committed:
Merge branch 'dev' into feature/#29_RightBox_Layout

sin-hyunjin commented:
### summary.js  

<img width="402" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/352c79ee-3d2c-44f3-a390-e0e384d45dca">
<img width="380" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/1e923ed8-33d8-447d-be22-e7ab05cd4bbb">

- tilte , textBox, rowText 다크모드 변경

### summaryButton.js 

<img width="395" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/cf949ef1-6f3f-43fe-acc3-4ac29660c1e1">
<img width="398" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/3bf763d3-11f4-4862-98c6-c7107ed92b99">

- title, textBox , rowText 다크모드변경 

### string.js 

<img width="407" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/10227a35-ea55-4a60-bf17-f5a24e6821be">
<img width="402" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/41557711-9a35-4985-a536-51c7c18e1b83">

- svg,text 다크모드변경 및 마우스 올렸을 시 애니메이션 추가 
- 화면이 1024px 이하일 때 버튼위치 수정 

### codeEditor.js , summaryRightBox.js

<img width="499" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/623416c5-c560-4ab1-89e3-d6e2755c5577">

- firstSummary 다크모드 적용
- 코드 색상 변경 
- 다크모드일시 rightBox 배경이미지 변경 

해야할일 
- secondSummary , firstSummaryButton, secondSummaryButton 코드색상 변경 
- rightBox.js 나머지부분 레이아웃 구성 
- rightBox 마우스이벤트 발생시 반응형으로 스타일변경할 것

sin-hyunjin committed:
DarkMode 적용

sin-hyunjin committed:
rightbox 위치변경

sin-hyunjin committed:
codeEditor.js 코드색상 다크모드변경완료

sin-hyunjin committed:
코드색상 클레스네임 변경

sin-hyunjin committed:
rightBox내부 비디오 부분 컴포넌트화 및 VideoList 레이아웃구성

sin-hyunjin commented:
summaryVideo.js 컴퍼넌트 추가 후 비디오부분 구성 
<img width="999" alt="image" src="https://github.com/doyou1/web-study-clone/assets/116487398/bbef8ced-b8b7-4d0c-acef-2114d2b63ffa">

sin-hyunjin committed:
rightBox Layout구성완료

sin-hyunjin committed:
(max-width: 800px) 일때 텍스트 위치수정

sin-hyunjin committed:
summaryRightBox.js 이미지 선택시 변경

ConfsSlug 부분 2021년, 2019년 선택시 이미지 변경되는 코드 작성

sin-hyunjin committed:
summaryButton 다크모드시 border 색상추가

sin-hyunjin committed:
Summary button -> rightBox 사진추가

doyou1 committed:
Merge branch 'dev' into feature/#29_RightBox_Layout
